src/configurationProvider.ts

```diff
@@ -203,6 +203,11 @@ export class JavaDebugConfigurationProvider implements vscode.DebugConfiguration
         return !Array.isArray(configItems) || !configItems.length;
     }
 
+    private isValidConfig(config: vscode.DebugConfiguration): boolean {
```

Comment:
The intention is not for checking if config is valid or not, but find the case that launch.json is missing or empty. So isEmptyConfig is more appropriate.

Reply:
> So isEmptyConfig is more appropriate.

After some consideration, I agree that `isValidConfig` might not be proper.  

Two reasons:
1. Confusion. `isValidConfig` implies the config contains all the information needed to launch a debug session, but actually it doesn't. 
2. For the name `isValidConfig`, it should have different value for following 2 configs, one is valid and the other invalid. However it doesn't, the function simply counts the keys and ignores "noDebug". 

```jsonc
// a valid config:
        {
            "type": "java",
            "name": "Debug (Launch)",
            "request": "launch", 
            "cwd": "${workspaceFolder}",
            "console": "internalConsole",
            "stopOnEntry": false,
            "mainClass": "",
            "args": ""
        }
// an invalid config:
       {
            "type": "java",
            "name": "Debug (Launch)",
            "request": "invalidValue",   // <------HERE
            "cwd": "${workspaceFolder}",
            "console": "internalConsole",
            "stopOnEntry": false,
            "mainClass": "",
            "args": ""
        }
```

For the name `isEmptyConfig`, yes we just regard `{noDebug: true}` as "empty", it's exactly why we need a helper function to customize the way we judge an empty config. Finally I'm ok with the name.
@yaohaizh are you ok with the name, or a better candidate if possible.